src/Microsoft.Extensions.ML/ServiceCollectionExtensions.cs

```diff
@@ -14,7 +15,7 @@ namespace Microsoft.Extensions.ML
     public static class ServiceCollectionExtensions
     {
         /// <summary>
-        /// Adds a <see cref="PredictionEnginePoolBuilder{TData, TPrediction}"/> to the service collection.
+        /// Adds a <see cref="PredictionEnginePool{TData, TPrediction}"/> and required config services to the service collection.
```

Comment:
This should stay `PredictionEnginePoolBuilder`, that is what is added to the service collection.

Reply:
Are you sure? The code seems pretty clear in that it adds a `PredictionEnginePool<TData, TPrediction>` object:

```cs
services.AddSingleton<PredictionEnginePool<TData, TPrediction>
```

What is returned is a `PredictionEnginePoolBuilder<TData, TPrediction>` object which simply wraps `IServiceCollection`. Speaking of which, this class doesn't do anything with the generic arguments which could probably even be removed?

Reply:
> Speaking of which, this class doesn't do anything with the generic arguments which could probably even be removed?

You can't change the generic arguments because this class is public and has shipped as a stable API.

Also, the generic arguments are used by the extension methods.